src/Widgets/DeviceItem.vala

```diff
@@ -137,6 +137,9 @@ namespace Network.Widgets {
                         subtitle = _("Enabled (auto mode)");
                         status_image.icon_name = "user-available";
                         break;
+                    default:
```

Comment:
I think we want to make sure we're actually handling all the possible cases here rather than just ignoring a warning

Reply:
I’m not sure what it is all about.

The `Utils.CustomMode` seems to contain unrelated things: the `PROXY_*` constants and 2 `HOTSPOT_*` constants. In the case of this `switch_status` function, it looks like being called only from 3 places:

- [Widgets/DeviceItem.vala:49 & 51](https://github.com/milouse/switchboard-plug-network/blob/remove-warnings/src/Widgets/DeviceItem.vala#L49)
- [Views/ProxyPage.vala:97](https://github.com/milouse/switchboard-plug-network/blob/remove-warnings/src/Views/ProxyPage.vala#L97)
- [Views/VPNPage.vala:170](https://github.com/milouse/switchboard-plug-network/blob/remove-warnings/src/Views/VPNPage.vala#L170)

None of these call are dealing with `HOTSPOT_*` related stuff.

That’s why I proposed to silently skip possibilities which should never occur. However a more robust/clean solution would be to remove the `HOTSPOT_*` constants from the `CustomMode` enum if we are not using them anymore? It looks like those constants have been forgotten in https://github.com/elementary/switchboard-plug-network/pull/176

And if we remove them from `CustomMode`, maybe we should rename this enum to `ProxyMode`?

What do you think?

Reply:
Ah okay yeah, that makes sense since hotspot is handled already we should remove those from `CustomMode` and I agree makes sense to rename this to `ProxyMode`